pyaedt/modeler/cad/Primitives.py

```diff
@@ -275,45 +587,4908 @@ def version(self):
+                new_object_id_dict[obj.name] = old_id
+                new_object_dict[old_id] = obj
+        for old_id, obj in self.points.items():
+            if obj.name in self._points:
+                new_points_dict[obj.name] = obj
+        self.objects = new_object_dict
+        self._object_names_to_ids = new_object_id_dict
+        self.points = new_points_dict
+
+    @pyaedt_function_handler()
+    def find_new_objects(self):
+        """Find any new objects in the modeler that were created
+        by previous operations.
+
+        Returns
+        -------
+        dict
+            Dictionary of new objects.
+
+        """
+        new_objects = []
+        for obj_name in self.object_names:
+            if obj_name not in self._object_names_to_ids:
+                new_objects.append(obj_name)
+        return new_objects
+
+    @pyaedt_function_handler()
+    def add_new_objects(self):
+        """Add objects that have been created in the modeler by
+        previous operations.
+
+        Returns
+        -------
+        list
+            List of added objects.
+
+        """
+        # TODO: Need to improve documentation for this method.
+        added_objects = []
+
+        for obj_name in self.object_names:
+            if obj_name not in self._object_names_to_ids:
+                self._create_object(obj_name)
+                added_objects.append(obj_name)
+        for obj_name in self.unclassified_names:
+            if obj_name not in self._object_names_to_ids:
+                self._create_object(obj_name)
+                added_objects.append(obj_name)
+        for obj_name in self.point_names:
+            if obj_name not in self.points.keys():
+                self._create_object(obj_name)
+                added_objects.append(obj_name)
+        return added_objects
+
+    @pyaedt_function_handler()
+    def add_new_user_defined_component(self):
+        """Add 3D components and user-defined models that have been created in the modeler by
+        previous operations.
+
+        Returns
+        -------
+        list
+            List of added components.
+
+        """
+        added_component = []
+        for comp_name in self.user_defined_component_names:
+            if comp_name not in self.user_defined_components:
+                self._create_user_defined_component(comp_name)
+            added_component.append(comp_name)
+        return added_component
+
+    # TODO Eliminate this - check about import_3d_cad
+    # Should no longer be a problem
+    @pyaedt_function_handler()
+    def refresh_all_ids(self):
+        """Refresh all IDs."""
+
+        self.add_new_objects()
+        self.add_new_user_defined_component()
+        self.cleanup_objects()
+
+        return len(self.objects)
+
+    @pyaedt_function_handler()
+    def get_objects_by_material(self, materialname=None):
+        """Retrieve a list of objects either of a specified material or classified by material.
```

Comment:
```suggestion
        """Get a list of objects either of a specified material or classified by material.
```
It seems more intutitve to use the same simple verb used in the method name.